experimental/dds/tree2/src/feature-libraries/detachedFieldIndexSummarizer.ts

```diff
@@ -25,15 +25,18 @@ const detachedFieldIndexBlobKey = "DetachedFieldIndexBlob";
 export class DetachedFieldIndexSummarizer implements Summarizable {
 	public readonly key = "DetachedFieldIndex";
 
-	public constructor(private readonly detachedFieldIndex: DetachedFieldIndex) {}
+	public constructor(private readonly getDetachedFieldIndex: () => DetachedFieldIndex) {}
```

Comment:
this is pretty ugly. Maybe using `ICachedValue<DetachedFieldIndex>` is conceptually nicer? It suggests the index can be 'invalidated'.

Reply:
similar concern for forest.

Reply:
kinda resolved in latest. it's a bit weird that the set of summarizables can change still, but less weird than here.